### DISCUSSION TITLE
win-vind v5.0 beta test

### DISCUSSION THREAD

**Discussion by pit-ray** [pit-ray/win-vind#129] [Development] [answered]
Thank you for your patience.

I have fundamentally rewritten the entire core key mapping system because win-vind has long maintained a complex and unmanageable mapping system and I had been unable to respond to many issues.

This makes it almost incompatible with the previous win-vind, but it allows for many things and is extremely simple to set up.

However, since the changes are extremely extensive, I cannot find the bugs completely on my own.

If anyone would like to contribute to the development, please try the new win-vind 5.0.0-beta from the link below and report any strange behavior in this thread. **Please include reproduction steps in your report.** Pull requests to master are also fine.

If you are participating, please read the change log in the release carefully.

~~https://github.com/pit-ray/win-vind/releases/tag/v5.0.0-beta1~~
~~https://github.com/pit-ray/win-vind/releases/tag/v5.0.0-beta2~~
~~https://github.com/pit-ray/win-vind/releases/tag/v5.0.0-beta3~~
~~https://github.com/pit-ray/win-vind/releases/tag/v5.0.0-beta4~~
https://github.com/pit-ray/win-vind/releases/tag/v5.0.0-beta5

## ToDo List for production release
- [x] Integration test
- [x] The documents for v5 in [v5docs](https://github.com/pit-ray/win-vind/tree/v5docs) branch.

**Comment by pit-ray** [comment]
When you input the command consisted only system key (e.g. `<esc-right>`), the parser will jam with them in beta1.
I fixed the problem as beta2.

Please try it instead of beta1.
~~https://github.com/pit-ray/win-vind/releases/tag/v5.0.0-beta2~~  
https://github.com/pit-ray/win-vind/releases/tag/v5.0.0-beta3

**Comment by gilbh** [comment]
Just to make sure: I downloaded the chocolatey installer and copy-pasted the files in `C:\ProgramData\chocolatey\bin\`

The About shows 5.0.0. Is this indeed 5.0.2?

**Reply by pit-ray** [reply]
It's partially correct. This version is v5.0.0-beta2, so it shows 5.0.0.

win-vind adopts `[major].[minor].[patch]` style version format.
If the production release's 5.0.0 has some bugs and we fix them, it will be 5.0.1.

**Comment by gilbh** [comment]
I see. So I guess the portable version is the latest patch of v5.0.2

I will work with that.

**Reply by pit-ray** [reply]
Anyway, as long as the download source is correct, you should be fine.
The **About**'s version is not for the beta version, only for the production version.

**Comment by pit-ray** [comment]
I have released beta3.
https://github.com/pit-ray/win-vind/releases/tag/v5.0.0-beta3

In beta3, I have deployed scripts in [tests/runtime](https://github.com/pit-ray/win-vind/tree/master/tests/runtime) directory to automatically run the test in Python. If anyone can write Python, it would be helpful to add test cases (refer to the [sample codes](https://github.com/pit-ray/win-vind/blob/master/tests/runtime/checks/mouse.py)). To execute the test, please read documents in [tests/README.md](https://github.com/pit-ray/win-vind/tree/master/tests).

I plan test cases for each function like the following. Ideally, one case per function.
- Mouse movement
- Mouse jumping
- Window operation (create your own GUI in Python using TK or Qt and get parameters about the window.)
- Editor emulation. (create your own text area in Python to get cursor position)

I plan the target release date to be the end of this month.

Thanks.

**Comment by pit-ray** [comment]
I released the beta4, please check it.
https://github.com/pit-ray/win-vind/releases/tag/v5.0.0-beta4

**Comment by pit-ray** [comment]
v5 will be a modern icon like the following!

### v1 ~ v4
<img src="https://user-images.githubusercontent.com/42631247/213970672-d679f08f-0b68-4017-a4a8-8be0380c9357.png" width = 128>

### v5
<img src="https://user-images.githubusercontent.com/42631247/213970563-30cb5dcb-1ca0-4c5f-9109-dd95c49192e6.png" width=128>


![image](https://user-images.githubusercontent.com/42631247/213971065-0c9a1393-6f78-4300-96af-263379fe79e1.png)
![image](https://user-images.githubusercontent.com/42631247/213971362-c08b2c6a-8003-4272-9a02-2187ffd1b5d9.png)

**Comment by pit-ray** [comment]
I released beta5 including new icon.
https://github.com/pit-ray/win-vind/releases/tag/v5.0.0-beta5

**Comment by pit-ray** [comment, marked as answer]
Release today. Thank you for beta testing.
